### PR TITLE
Widecone headlight

### LINKED ISSUE #21874: Add Wide-angle headlights and directed floodlights as vehicle parts.
It seems that something that would satisfy the request in #21852 better than dropping floodlight power consumption is a "vehicle floodlight with 90deg or 180deg range and reasonable power consumption".

As such I propose adding a "Wide-angle Headlight" (with a 90 degree arc and -375 epower) and a "Directional Floodlight" (with a 180 degree arc and -750 epower) as vehicle parts.

The light cones are applied here: https://github.com/CleverRaven/Cataclysm-DDA/blob/master/src/lightmap.cpp#L338

The locations where CONE_LIGHT are referenced: https://github.com/CleverRaven/Cataclysm-DDA/search?utf8=%E2%9C%93&q=CONE_LIGHT need to be updated, either by adding variants of VPART_CONE_LIGHT or by making the light cone flag include a variable indicating the angle that the light illuminates.

Also it requires the vehicle parts, related items and crafting recipes for them, which are going to be nearly identical to the existing headlight items.

### PR DESCRIPTION
**Pull request by peperov**
#### Summary
```SUMMARY: Content "Adds 'wide-angle headlight', 'reinforced wide-angle headlight' and 'directed floodlight'"``` 

#### Purpose of change
Fixes #21874 - added wide-angle headlight (90 degrees, -375e) and directed floodlight (180 degrees, -750e) including crafting recipes and vehicle parts.

#### Describe the solution
Three new vehicle parts were added:
wide_headlight (90 degrees, -375e)
wide_headlight_reinforced (90 degrees, -375e)
directed_floodlight (180 degrees, -750e)

Three new craftable items were added:
car_wide_headlight
wide_headlight_reinforced
directed_floodlight

Naming conventions follow existing items and parts.

Added Keybindings for each item group:
headlights ("h", unchanged)
wide-angle headlights ("b", new)
directed floodlights ("O", new)
floodlights ("o", unchanged)

#### Describe alternatives you've considered

I considered to merge the keybindings of headlights and wide-angle headlights, but decided against it so the user can have differing lighting setups for differing situations (given the new lights allow for very situational application). I also considered to merge the keybindings of the directed floodlight and the regular floodlight, but decided against it for the same reasoning.
I considered to have a regular car headlight be installable as wide-angle headlight and a regular floodlight be installable as a directed floodlight, both for simplification. I decided against them because the issue explicitly mentions new items and recipes.
I considered naming the 'directed floodlight' as 'directional floodlight' but decided against it. If 'directional' is better english in that context, please let me know.

#### Additional context

This is my first contribution to the project. I still have trouble with the build pipeline (astyle, tests, jenkins).

I tested each added feature thoroughly: crafting the items, adding them to a vehicle, the light cones and the power drain.

I did not add the new items to the existing tilesets, but that can be done. I wanted to await your critics and suggestions before doing so.

Some screenshots:
directed overhead
overhead
wide angle headlight
normal headlight

### CODE REVIEW

**Comment by peperov** [comment]
One question for my personal information:
How can the gorgon build finish but fail? Did it not compile?

Also: I encountered a problem on my Ubuntu machine. When I use the premade build script to build CDDA, it fails saying:
'make: build-scripts/validate_pr_in_jenkins: Command not found'
'Makefile:1101: recipe for target 'validate-pr' failed'
Anyone got some ideas on this? Could it be related to me trying to change my java sdk? Anyone seen this before?

**Comment by FrozenSnowFox** [comment]
Nice addition to the game.  Is there any specific reason floodlights (normal and directed) can't be reinforced?  It's the one thing that always breaks on my mobile base when it gets attacked on the sides.

File: src/veh_interact.cpp [hunk 1]
```diff
@@ -2658,13 +2660,17 @@ void veh_interact::complete_vehicle()
...
+             vpinfo.has_flag( VPFLAG_HALF_CIRCLE_LIGHT ) ) {
             // Stash offset and set it to the location of the part so look_around will start there.
             int px = g->u.view_offset.x;
             int py = g->u.view_offset.y;
             g->u.view_offset.x = veh->global_pos3().x + q.x - g->u.posx();
             g->u.view_offset.y = veh->global_pos3().y + q.y - g->u.posy();
-            popup(_("Choose a facing direction for the new headlight.  Press space to continue."));
+
+            bool is_overheadlight = vpinfo.has_flag( VPFLAG_HALF_CIRCLE_LIGHT );
+            popup(_("Choose a facing direction for the new %s.  Press space to continue."), is_overheadlight ? "overhead light" : "headlight");
```
Comment (mlangsdorf) [inline src/veh_interact.cpp]:
```suggestion
            popup( _( "Choose a facing direction for the new %s.  Press space to continue." ), is_overheadlight ? "overhead light" : "headlight" );
```

Comment (peperov) [reply src/veh_interact.cpp]: I'll add it in.

**Comment by peperov** [comment]
> Nice addition to the game. Is there any specific reason floodlights (normal and directed) can't be reinforced? It's the one thing that always breaks on my mobile base when it gets attacked on the sides.

It has not been in the game ever I think. Just don't mount the floodlights on the outside of the car (same with solar panels and vehicle tanks). Or make an issue and link me to it.

**Comment by peperov** [comment]
Should I add the new vehicle parts to the Tilesets?
Do I also need to add the items to Tilesets?

File: src/game.cpp [hunk 2]
```diff
@@ -1357,7 +1357,8 @@ static int veh_lumi( vehicle &veh )
 
     for( const auto pt : lights ) {
         const auto &vp = pt->info();
-        if( vp.has_flag( VPFLAG_CONE_LIGHT ) ) {
+        if( vp.has_flag( VPFLAG_CONE_LIGHT ) ||
```
Comment (AMurkin) [inline src/game.cpp]:
```suggestion
        if( vp.has_flag( VPFLAG_CONE_LIGHT ) ||
```

**Comment by AMurkin** [comment]
You can use "looks_like" JSON field.

**Comment by peperov** [comment]
Thank you that is a great hint.

**Pull request merged**